tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -187,7 +187,7 @@ static PyObject * THPVariable_range(PyObject* self, PyObject* args, PyObject* kw
 
   if (r.idx == 0) {
     PyErr_WarnEx(PyExc_UserWarning, "torch.range is deprecated in favor of torch.arange "
-        "and will be removed in 0.5. Note that arange generates values in [start; end), "
+        "and will be removed in a future release. Note that arange generates values in [start; end), "
```

Comment:
Updating this warning message is a good idea. I had the opportunity to review the history of this deprecation with @gchanan and we'd like to clarify why the function is being deprecated too, and that's because it's inconsistent with Python's range function (see the ancient https://github.com/pytorch/pytorch/issues/733).

What about a sentence like, "torch.range is deprecated and will be removed in a future release because its behavior is inconsistent with Python's range builtin. Use torch.arange, which produces values in [start, end), instead."

We could also update the warning in the documentation for `torch.range` to something like:

"This function is deprecated and will be removed in a future release because its behavior is inconsistent with Python's range builtin. Use torch.arange, which produces values in [start, end), instead."

Do you think that would improve these warnings, @xuhdev?

Reply:
Definitely, I agree the new message is more informative. In addition, I also added that `torch.arange` runs faster as an additional incentive. (`arange` is vectorized but `range` is not)